### PR TITLE
FLOAT support for Parquet

### PR DESCRIPTION
**Pull request by cawallin**

### CODE REVIEW

**Comment by cawallin** [comment]
The Parquet predicate pushdown test is strictly for correctness (and isn't even extensive in the correctness testing), but I didn't know the best way to test it -- since none of the Parquet pushdown is tested

File: presto-hive/src/main/java/com/facebook/presto/hive/parquet/predicate/TupleDomainParquetPredicate.java [hunk 1]
```diff
@@ -157,26 +159,29 @@ else if (type.equals(BIGINT) && (statistics instanceof LongStatistics || statist
...
             return createDomain(type, hasNullValue, parquetDoubleStatistics);
         }
+        else if (type.equals(FLOAT) && statistics instanceof FloatStatistics) {
+            FloatStatistics floatStatistics = (FloatStatistics) statistics;
+            // ignore corrupted statistics
+            if (floatStatistics.genericGetMin() > floatStatistics.genericGetMax()) {
+                return Domain.create(ValueSet.all(type), hasNullValue);
+            }
+
+            ParquetIntegerStatistics parquetStatistics = new ParquetIntegerStatistics(
```
Comment (maciejgrzybek) [inline presto-hive/src/main/java/com/facebook/presto/hive/parquet/predicate/TupleDomainParquetPredicate.java:178]: Shouldn't we use `FloatStatistics`? Operating on bit representation is meaningless for `TupleDomain`. Especially in context of e.g. `createDomain` where we create ranges based on these values.

File: presto-hive/src/test/java/com/facebook/presto/hive/parquet/AbstractTestParquetReader.java [hunk 2]
```diff
@@ -176,7 +177,21 @@ private void testRoundTripNumeric(Iterable<Integer> writeValues)
...
+    {
+        tester.testRoundTrip(javaFloatObjectInspector, ImmutableList.of(1000.0f, -1.0f, Float.POSITIVE_INFINITY), FLOAT);
+        tester.testRoundTrip(javaFloatObjectInspector, ImmutableList.of(-1000.0f, Float.NEGATIVE_INFINITY, 1.0f), FLOAT);
+        tester.testRoundTrip(javaFloatObjectInspector, ImmutableList.of(0.0f, Float.NEGATIVE_INFINITY, Float.POSITIVE_INFINITY), FLOAT);
+
+        tester.testRoundTrip(javaFloatObjectInspector, ImmutableList.of(Float.NaN, -1.0f, 1.0f), FLOAT);
+        tester.testRoundTrip(javaFloatObjectInspector, ImmutableList.of(Float.NaN, -1.0f, Float.POSITIVE_INFINITY), FLOAT);
+        tester.testRoundTrip(javaFloatObjectInspector, ImmutableList.of(Float.NaN, Float.NEGATIVE_INFINITY, 1.0f), FLOAT);
+        tester.testRoundTrip(javaFloatObjectInspector, ImmutableList.of(Float.NaN, Float.NEGATIVE_INFINITY, Float.POSITIVE_INFINITY), FLOAT);
     }
```
Comment (maciejgrzybek) [inline presto-hive/src/test/java/com/facebook/presto/hive/parquet/AbstractTestParquetReader.java:195]: What about fractions? E.g. `1.23`?

File: presto-hive/src/test/java/com/facebook/presto/hive/parquet/TestTupleDomainParquetPredicate.java [hunk 3]
```diff
@@ -109,4 +112,28 @@ private static BinaryStatistics stringColumnStats(String minimum, String maximum
         statistics.setMinMax(Binary.fromString(minimum), Binary.fromString(maximum));
         return statistics;
     }
+
+    @Test
+    public void testFloat()
```
Comment (maciejgrzybek) [inline presto-hive/src/test/java/com/facebook/presto/hive/parquet/TestTupleDomainParquetPredicate.java:117]: I think this test is not meaningful. It's testing implementation, not behavior.

Comment (cawallin) [inline presto-hive/src/test/java/com/facebook/presto/hive/parquet/TestTupleDomainParquetPredicate.java:117]: I agree, but at least it documents the weird expectations we have for float. all of the tests in this class are of a similar ilk.

Comment (maciejgrzybek) [inline presto-hive/src/test/java/com/facebook/presto/hive/parquet/TestTupleDomainParquetPredicate.java:117]: This is weird but becomes a standard. I won't insist about this test though.

File: presto-hive/src/main/java/com/facebook/presto/hive/parquet/predicate/TupleDomainParquetPredicate.java (hunk 1 above)
Comment (cawallin) [inline presto-hive/src/main/java/com/facebook/presto/hive/parquet/predicate/TupleDomainParquetPredicate.java:178]: Weirdly, the tests pass when pushdown is enabled like this. I'll look a little more carefully into this.

Comment (cawallin) [inline presto-hive/src/main/java/com/facebook/presto/hive/parquet/predicate/TupleDomainParquetPredicate.java:178]: I've just confirmed that the TupleDomain contains the float-to-raw-int-bits Long value. This seems not very user friendly, but I don't know how we can change that without special casing everything for float.

Comment (maciejgrzybek) [inline presto-hive/src/main/java/com/facebook/presto/hive/parquet/predicate/TupleDomainParquetPredicate.java:178]: That's sad... Thanks for confirming this!

**Comment by cawallin** [comment]
Included in #229

**Pull request closed without merging**